### PR TITLE
Migrate from CircleCI to Github Actions

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
        The PR build (CI) is a reusable [required workflow](https://github.com/mattermost/actions-workflows/tree/main/.github/workflows/required_workflows/plugins)  

        #### Ticket Link
        Ticket: https://mattermost.atlassian.net/browse/CLD-4816

### CODE REVIEW

**Review by hanzei — CHANGES_REQUESTED** [review]
A few questions from my side before we migrate a major repository to GitHub actions:
1. Did you test the Plugin release process with GitHub actions? If I remember correctly, there was an issue with the GitLab release.
2. Do you know if the coverage reports still work?
3. Why is each job takes so long? In https://github.com/mattermost/mattermost-plugin-github/actions/runs/3901202765/jobs/6662764665 npm install took 7 minutes alone. The caching of `actions/setup-node` does not seem to work. Is the caching branch specific?
4. It seems like go modules are not getting cached. We could speed up the build time by doing that.
5. There is a warning about two deprecated uses in `aws-actions/configure-aws-credentials`. We should use the `v1-node16` branch of  `aws-actions/configure-aws-credentials`. (See https://github.com/aws-actions/configure-aws-credentials/issues/489#issuecomment-1278145876)
6. What happened to the nightly build? It helped spot flaky tests and dependency issues.
7. Why is installing Go not part of `mattermost/actions/plugin-ci/setup`? https://github.com/mattermost/actions-workflows/blob/5fc28f840725ffd21bcd580622a5fb3f3ef1dd78/.github/workflows/plugin-ci.yml#L58-L62

**Comment by spirosoik** [comment]
> A few questions from my side before we migrate a major repository to GitHub actions:
> 
> 1. Did you test the Plugin release process with GitHub actions? If I remember correctly, there was an issue with the GitLab release.
> 2. Do you know if the coverage reports still work?
> 3. Why is each job takes so long? In https://github.com/mattermost/mattermost-plugin-github/actions/runs/3901202765/jobs/6662764665 npm install took 7 minutes alone. The caching of `actions/setup-node` does not seem to work. Is the caching branch specific?
> 4. It seems like go modules are not getting cached. We could speed up the build time by doing that.
> 5. There is a warning about two deprecated uses in `aws-actions/configure-aws-credentials`. We should use the `v1-node16` branch of  `aws-actions/configure-aws-credentials`. (See [aws-actions/configure-aws-credentials#489 (comment)](https://github.com/aws-actions/configure-aws-credentials/issues/489#issuecomment-1278145876))
> 6. What happened to the nightly build? It helped spot flaky tests and dependency issues.
> 7. Why is installing Go not part of `mattermost/actions/plugin-ci/setup`? https://github.com/mattermost/actions-workflows/blob/5fc28f840725ffd21bcd580622a5fb3f3ef1dd78/.github/workflows/plugin-ci.yml#L58-L62

1. We have fixed the issue and it's tested.
2. It does, in couple of other repos I created PRs it has been reported correctly back in PR. Codecov successfully has been sent https://github.com/mattermost/mattermost-plugin-jira/actions/runs/4085912143/jobs/7044518837#step:5:1059  . I am wondering if it's something specific for this repo only
3. For node seems to be working in the logs, I think the `npm` dependencies is the issue, I will cross-check this, but it's a separate problem which we can fix after as the CI is centralised
4. They are check here https://github.com/mattermost/mattermost-plugin-jira/actions/runs/4085912143/jobs/7044642206#step:4:35
5.  Again we can fix this after the first migration as this is an on-going process but first we need to move everything to Github Actions.
6. We can add this back for sure. Again we will do this as a separate task
7. We can add it under setup too

**Comment by spirosoik** [comment]
@hanzei actually for 3, the first time indeed the cache is missing but if you see the build the npm is restored from the cache.

**Comment by spirosoik** [comment]
@hanzei I haven't seen any nightly builds in the orb or in this workflow of `.cricleci`. Where are these defined?

**Comment by hanzei** [comment]
> @hanzei I haven't seen any nightly builds in the orb or in this workflow of `.cricleci`. Where are these defined?

https://github.com/mattermost/mattermost-plugin-demo/blob/d9c54d6a91cecdd9d2d7571d074d3d1e368b6c79/.circleci/config.yml#L8-L19

**Comment by spirosoik** [comment]
ok I will create a ticket to support this separately.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "^v.*"
+  pull_request:
+
+permissions:
+  contents: read
```
Comment (mickmister) [inline .github/workflows/ci.yml:11]: I think we're good on limiting CI runs this way

**Review by mickmister — APPROVED** [review]
LGTM

**Comment by mickmister** [comment]
For slow `npm install`s, we used this with CircleCI to decrease download time of dependencies https://github.com/mattermost/circleci-orbs/blob/ea9f34540218ce3a62b101eab2c4fffc331be11b/plugin-ci/orb.yml#L46

```
git config --global url."ssh://git@".insteadOf git://
```

Maybe it would help here too?

**Comment by spirosoik** [comment]
> For slow `npm install`s, we used this with CircleCI to decrease download time of dependencies https://github.com/mattermost/circleci-orbs/blob/ea9f34540218ce3a62b101eab2c4fffc331be11b/plugin-ci/orb.yml#L46
> 
> ```
> git config --global url."ssh://git@".insteadOf git://
> ```
> 
> Maybe it would help here too?

I don't see any issue in terms of caching the lint and test after the first time when everything cache works pretty fast.

**Comment by spirosoik** [comment]
@hanzei done

**Review by hanzei — APPROVED** [review]
LGTM :rocket:

**Pull request merged**